lib/ClangImporter/SwiftDeclSynthesizer.cpp

```diff
@@ -2534,128 +2534,167 @@ SwiftDeclSynthesizer::makeDefaultArgument(const clang::ParmVarDecl *param,
+
+  for (clang::CXXConstructorDecl *selectedCtorDecl : cxxRecordDecl->ctors()) {
+    if (selectedCtorDecl->isDeleted() ||
+        selectedCtorDecl->getAccess() == clang::AS_private ||
+        selectedCtorDecl->getAccess() == clang::AS_protected ||
+        selectedCtorDecl->isCopyOrMoveConstructor())
+      continue;
+
+    unsigned int ctorParamCount = selectedCtorDecl->getNumParams();
+    std::string funcName = "__returns_" + cxxRecordDecl->getNameAsString();
+    if (ctorParamCount > 0)
+      funcName += "_" + std::to_string(ctorParamCount) + "_params";
+    clang::IdentifierInfo *funcNameToSynthesize = &clangIdents.get(funcName);
+
+    llvm::SmallVector<clang::QualType, 4> paramTypes;
+    for (const auto *param : selectedCtorDecl->parameters())
+      paramTypes.push_back(param->getType());
+    clang::FunctionProtoType::ExtProtoInfo EPI;
+    clang::QualType funcTypeToSynthesize =
+        clangCtx.getFunctionType(cxxRecordPtrTy, paramTypes, EPI);
+
+    clang::CXXMethodDecl *synthesizedCxxMethodDecl =
+        clang::CXXMethodDecl::Create(
+            clangCtx, const_cast<clang::CXXRecordDecl *>(cxxRecordDecl),
+            cxxRecordDeclLoc,
+            clang::DeclarationNameInfo(funcNameToSynthesize, cxxRecordDeclLoc),
+            funcTypeToSynthesize,
+            clangCtx.getTrivialTypeSourceInfo(funcTypeToSynthesize),
+            clang::SC_Static, /*UsesFPIntrin=*/false, /*isInline=*/true,
+            clang::ConstexprSpecKind::Unspecified, cxxRecordDeclLoc);
+    assert(
+        synthesizedCxxMethodDecl &&
+        "Unable to synthesize static factory for c++ foreign reference type");
+    synthesizedCxxMethodDecl->setAccess(clang::AccessSpecifier::AS_public);
+
+    llvm::SmallVector<clang::ParmVarDecl *, 4> synthesizedParams;
+    for (unsigned int i = 0; i < ctorParamCount; ++i) {
+      auto *origParam = selectedCtorDecl->getParamDecl(i);
+      auto *param = clang::ParmVarDecl::Create(
+          clangCtx, synthesizedCxxMethodDecl, cxxRecordDeclLoc,
+          cxxRecordDeclLoc, origParam->getIdentifier(), origParam->getType(),
+          clangCtx.getTrivialTypeSourceInfo(origParam->getType()),
+          clang::SC_None, nullptr);
+      synthesizedParams.push_back(param);
+    }
+    synthesizedCxxMethodDecl->setParams(synthesizedParams);
+
+    if (!hasImmortalAttrs(cxxRecordDecl)) {
+      clang::SwiftAttrAttr *returnsRetainedAttrForSynthesizedCxxMethodDecl =
+          clang::SwiftAttrAttr::Create(clangCtx, "returns_retained");
+      synthesizedCxxMethodDecl->addAttr(
+          returnsRetainedAttrForSynthesizedCxxMethodDecl);
+    }
+
+    std::string swiftInitStr = "init(";
+    for (unsigned i = 0; i < ctorParamCount; ++i) {
+      swiftInitStr += "_:";
+    }
+    swiftInitStr += ")";
+    clang::SwiftNameAttr *swiftNameAttr =
+        clang::SwiftNameAttr::Create(clangCtx, swiftInitStr);
+    synthesizedCxxMethodDecl->addAttr(swiftNameAttr);
+
+    llvm::SmallVector<clang::Expr *, 8> ctorArgs;
```

Comment:
Other small vectors had 4 as their inline buffer size, this one has 8. I think we should pick one and stick to that in the whole function unless we have a reason to diverge.

Reply:
This was mistakenly marked as resolved. I still see different values for the buffer sizes.